test/CMakeLists.txt

```diff
@@ -67,6 +70,18 @@ macro(onedpl_add_test test_source_file)
         target_compile_options(${_test_name} PRIVATE /bigobj)
     endif()
 
+    # Disable checked iterators on Windows for debug mode
+    # For details please see
+    #    https://learn.microsoft.com/en-us/cpp/standard-library/iterator-debug-level?view=msvc-170
+    #    https://learn.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-170
+    #    https://stackoverflow.com/questions/51494506/replacing-md-with-mt-in-cmake-is-not-possible-in-release-mode
+    if (WIN32 AND DEFINED ${_build_type_in_lower} AND ${_build_type_in_lower} STREQUAL "debug")
```

Comment:
I tried testing this change and I believe this condition is not correct. As written, you are not checking if the variable `_build_type_in_lower` exists, but rather if there exists a variable whose name is provided by the value of `${_build_type_in_lower}` (e.g., a variable with the name `debug` or `release`).

This condition will always be false and so the workaround is never actually applied.

I considered changing it to the following, but this doesn't make much sense as the variable `_build_type_in_lower ` is always defined since it was explicitly defined at the beginning of the macro:

```suggestion
    if (WIN32 AND DEFINED _build_type_in_lower AND ${_build_type_in_lower} STREQUAL "debug")
```

Instead, I would suggest to define `_build_type_in_lower` as a known string that is not debug, and only redefine it if `CMAKE_BUILD_TYPE` is not falsey. Such as:

```diff
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 5b266617..bfb4cfb9 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -59,7 +59,10 @@ add_custom_target(run-onedpl-tests

 macro(onedpl_add_test test_source_file switch_off_checked_iterators)

-    string(TOLOWER "${CMAKE_BUILD_TYPE}" _build_type_in_lower)
+    set(_build_type_in_lower "unknown")
+    if (CMAKE_BUILD_TYPE)
+        string(TOLOWER "${CMAKE_BUILD_TYPE}" _build_type_in_lower)
+    endif()

     get_filename_component(_test_name ${test_source_file} NAME)
     string(REPLACE "\.cpp" "" _test_name ${_test_name})
@@ -75,11 +78,9 @@ macro(onedpl_add_test test_source_file switch_off_checked_iterators)
     #    https://learn.microsoft.com/en-us/cpp/standard-library/iterator-debug-level?view=msvc-170
     #    https://learn.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-170
     #    https://stackoverflow.com/questions/51494506/replacing-md-with-mt-in-cmake-is-not-possible-in-release-mode
-    if (WIN32 AND DEFINED ${_build_type_in_lower} AND ${_build_type_in_lower} STREQUAL "debug")
-        if (${switch_off_checked_iterators})
-            target_compile_definitions(${_test_name} PRIVATE _ITERATOR_DEBUG_LEVEL=0)
-            target_compile_options(${_test_name} PRIVATE "/MD$<$<CONFIG:Debug>:>")
-        endif()
+    if (WIN32 AND ${_build_type_in_lower} STREQUAL "debug" AND ${switch_off_checked_iterators})
+        target_compile_definitions(${_test_name} PRIVATE _ITERATOR_DEBUG_LEVEL=0)
+        target_compile_options(${_test_name} PRIVATE "/MD$<$<CONFIG:Debug>:>")
     endif()

     # oneDPL test harness may initialize a C++ iterator using iterator with different type
```

I have tested the following configurations and I'm able to compile the `xpu_accumulate.pass.exe` target for all of them on Windows:
```
cmake -G "Ninja" -DCMAKE_CXX_COMPILER=icx -DONEDPL_BACKEND=dpcpp -DCMAKE_BUILD_TYPE=Debug ..
cmake -G "Ninja" -DCMAKE_CXX_COMPILER=icx -DONEDPL_BACKEND=dpcpp -DCMAKE_BUILD_TYPE="" ..
cmake -G "Ninja" -DCMAKE_CXX_COMPILER=icx -DONEDPL_BACKEND=dpcpp ..
```

Reply:
Thank you, @adamfidel. Good point. Fixed.